main/reference/zoe-api/zcfseat.md

```diff
@@ -0,0 +1,214 @@
+
+**Note**: You should not use **ZCFSeat.exit()** when exiting with an error. Use the method **[ZCFSeat.fail()](#azcfseat-fail-msg)** instead. 
+
+## aZCFSeat.fail(msg)
+   - **msg** **String**
+   - Returns: None.
+
+The **seat** exits, displaying the optional **msg** string, if there is one, on the console.
+This is equivalent to exiting, except that **exit** is successful while
+**ZCFSeat.fail()** signals an error occurred while processing the offer. The contract
+still gets its current **allocation** and the **seat** can no longer interact with the contract.
+Any other still open seats or outstanding promises and the contract instance continue.
+
+Agoric recommends you exit a seat with an error as follows
+
+```js
+throw seat.fail(Error('you did it wrong'));
+```
+
+## aZCFSeat.hasExited()
+  - Returns: **Boolean**
+
+Returns **true** if the **ZCFSeat** has exited, **false** if it is still active.
+
+## aZCFSeat.getAmountAllocated(keyword, brand)
+  - **keyword** **String**
+  - **brand** **[Brand](/reference/ertp-api/brand.md)**
+  - Returns: **[Amount](/reference/ertp-api/ertp-data-types.md#amount)**
+
+Returns the **Amount** from the part of the **[Allocation](./zoe-data-types.md#allocation)** that matches the
+*keyword* and *brand*. If the *keyword* is not in the **Allocation**, it
+returns an empty **Amount** of the *brand* argument. (After
+**ZCFSeat.exit()** has been called, it continues to report the final allocation balance, 
+which was transferred to a payout.)
+
+This is similar to the next method, **getCurrentAllocation()**. **getAmountAllocated()**
+gets the **Allocation** of one keyword at a time, while **getCurrentAllocation()** returns
+all the current **Allocations** at once.
+
+## aZCFSeat.getCurrentAllocation()
+  - Returns: **[Allocation](./zoe-data-types.md#allocation)**
+
+An **Allocation** is an **AmountKeywordRecord** of key-value pairs where
+the key is a keyword such as **Asset** or **Price** applicable to the
+contract. The value is an **amount** with its **value** and **brand**.
+
+**Allocations** represent the **[Amounts](/reference/ertp-api/ertp-data-types.md#amount)** to be paid
+out to each **seat** on exit. (After
+**exit()** has been called, the final allocation balances, which were transferred to
+payouts, continue to be reported.)
+Normal reasons for exiting are the user requesting to exit or the contract
+explicitly choosing to close out the **seat**. The guarantees also hold if the contract
+encounters an error or misbehaves. There are several methods for finding out
+what **amount** a current **allocation** is.
+
+This is similar to the previous method, **getAmountAllocated()**. **getAmountAllocated()**
+gets the **allocation** of one keyword at a time, while **getCurrentAllocation()** returns
+all the current **allocations** at once.
+
+An **Allocation** example:
+```js
+{
+  Asset: AmountMath.make(quatloosBrand, 5n),
```

Comment:
The keywords `Asset` and `Price` are repeated throughout these docs. They seem kinda confusing to me. What do the chosen keywords mean to convey?

Reply:
Yeah, I found them kinda confusing as well. I plan on renaming most of them when I go through & spend time on the code snippets themselves.

To directly answer your question, the keywords convey whatever the author of the contract wants them to convey. They're just labels for whatever values appear after the colon.